utilities_common/multi_asic.py

```diff
@@ -0,0 +1,152 @@
+from utilities_common.constants import BGP_NEIGH_OBJ
+from utilities_common.constants import PORT_CHANNEL_OBJ
+from utilities_common.constants import PORT_OBJ
+
+
+class MultiAsic(object):
+
+    def __init__(self, display_option=DISPLAY_ALL, namespace_option=None):
+        self.namespace_option = namespace_option
+        self.display_option = display_option
+        self.current_namespace = None
+        self.is_multi_asic = is_multi_asic()
+
+    def is_object_internal(self, object_type, cli_object):
+        '''
+        The function checks if a CLI object is internal and returns true or false.
+        Internal objects are port or portchannel which are connected to other 
+        ports or portchannels within a multi ASIC device.
+
+        For single asic, this function is not applicable
+        '''
+        if object_type == PORT_OBJ:
+            return is_port_internal(cli_object)
+        elif object_type == PORT_CHANNEL_OBJ:
+            return is_port_channel_internal(cli_object)
+        elif object_type == BGP_NEIGH_OBJ:
+            return is_bgp_session_internal(cli_object)
+
+    def skip_display(self, object_type, cli_object):
+        '''
+        The function determines if the passed cli_object has to be displayed or not.
+        returns true if the display_option is external and  the cli object is internal.
+        returns false, if the cli option is all or if it the platform is single ASIC.
+
+        '''
+        if not self.is_multi_asic:
+            return False
+        if self.display_option == DISPLAY_ALL:
+            return False
+        return self.is_object_internal(object_type, cli_object)
+
+    def get_ns_list_based_on_options(self):
+        ns_list = []
+        if not self.is_multi_asic:
+            return [DEFAULT_NAMESPACE]
+        else:
+            namespaces = get_all_namespaces()
+            if self.namespace_option is None:
+                if self.display_option == DISPLAY_ALL:
+                    ns_list = namespaces['front_ns'] + namespaces['back_ns']
+                else:
+                    ns_list = namespaces['front_ns']
+            else:
+                if self.namespace_option not in namespaces['front_ns'] and \
+                        self.namespace_option not in namespaces['back_ns']:
+                    raise ValueError(
+                        'Unknown Namespace {}'.format(self.namespace_option))
+                ns_list = [self.namespace_option]
+        return ns_list
+
+
+def multi_asic_ns_choices():
+    if not is_multi_asic():
+        return [DEFAULT_NAMESPACE]
+    choices = get_namespace_list()
+    return choices
+
+
+def multi_asic_display_choices():
+    if not is_multi_asic():
+        return [DISPLAY_ALL]
+    else:
+        return [DISPLAY_ALL, DISPLAY_EXTERNAL]
+
+
+def multi_asic_display_default_option():
+    if not is_multi_asic():
+        return DISPLAY_ALL
+    else:
+        return DISPLAY_EXTERNAL
+
+
+_multi_asic_click_options = [
```

Comment:
I"m not sure how this works, is it possible to not display these options on single ASIC platforms.

Reply:
I believe Click 7.0 added a `hidden` keyword. However, Click 6.x (which is what we have in 201911) does not have support for this, which would make backporting a bit tedious. I believe @judyjoseph was working with it recently.